src/iOS/toga_iOS/widgets/scrollcontainer.py

```diff
@@ -19,86 +17,110 @@
             # pad the scrollview for the statusbar offset
-            padding_vertical = padding_vertical + self.__current_content.viewport.statusbar_height
+            padding_vertical = padding_vertical + self.current_content.viewport.statusbar_height
         else:
             content_height = self.native.frame.size.height
 
         self.native.setContentSize_(CGSizeMake(content_width + padding_horizontal,
                                                content_height + padding_vertical))
 
-    def __constrain_to_scrollview(self, widget):
+    def constrain_to_scrollview(self, widget):
         # The scrollview should know the content size as long as the
         # view contained has an intrinsic size and the constraints are
         # not ambiguous in any axis.
         view = widget.native
-        horizontal = NSLayoutConstraint.constraintsWithVisualFormat_options_metrics_views_(
-            'H:|[view]|',
-            0,
-            None,
-            {'view': view}
+        leading_constraint = NSLayoutConstraint.constraintWithItem_attribute_relatedBy_toItem_attribute_multiplier_constant_(
+            view,
+            NSLayoutAttributeLeading,
+            NSLayoutRelationEqual,
+            self.native,
+            NSLayoutAttributeLeading,
+            1.0,
+            0
         )
-        self.native.addConstraints_(horizontal)
-
-        vertical = NSLayoutConstraint.constraintsWithVisualFormat_options_metrics_views_(
-            'V:|[view]|',
-            0,
-            None,
-            {'view': view}
+        trailing_constraint = NSLayoutConstraint.constraintWithItem_attribute_relatedBy_toItem_attribute_multiplier_constant_(
+            self.native,
+            NSLayoutAttributeTrailing,
+            NSLayoutRelationEqual,
+            view,
+            NSLayoutAttributeTrailing,
+            1.0,
+            0
+        )
+        top_constraint = NSLayoutConstraint.constraintWithItem_attribute_relatedBy_toItem_attribute_multiplier_constant_(
+            view,
+            NSLayoutAttributeTop,
+            NSLayoutRelationEqual,
+            self.native,
+            NSLayoutAttributeTop,
+            1.0,
+            0
+        )
+        bottom_constraint = NSLayoutConstraint.constraintWithItem_attribute_relatedBy_toItem_attribute_multiplier_constant_(
+            self.native,
+            NSLayoutAttributeBottom,
+            NSLayoutRelationEqual,
+            view,
+            NSLayoutAttributeBottom,
+            1.0,
+            0
         )
-        self.native.addConstraints_(vertical)
+        self.native.addConstraints_([
+            leading_constraint,
+            trailing_constraint,
+            top_constraint,
+            bottom_constraint
+        ])
 
     def create(self):
+        self.current_content = None
+        self.vertical_enabled = True
+        self.horizontal_enabled = True
```

Comment:
Is there any reason this state needs to be duplicated on the implementation? Wouldn't it be possible to use the interface's representation (e.g., `self.interface.content._impl` for `current_content`)?

Reply:
Actually now that I look at it everything about the "state" I need is already in self.interface - I just wasn't familiar enough with the toga internals yet. I've removed these variables in 6ff5e83